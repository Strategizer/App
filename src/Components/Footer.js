import React from "react";
import Typography from '@material-ui/core/Typography';
import {Link} from "react-router-dom";

function Footer() {
  return (
    <div>  
      <span  style={{float : 'left'}}>
    <Typography variant="h6">
      <a style={{textDecoration: 'none'}} href="https://github.com/Strategizer/app">Open Source</a>
    </Typography></span>
      <span  style={{float : 'right'}}>
    <Typography variant="h6">
      <Link style={{textDecoration: 'none'}} to="/contributors">Contributors</Link>
    </Typography></span>
    </div>
  );
};

export default Footer;
