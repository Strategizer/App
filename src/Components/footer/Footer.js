import React from 'react';
import { Link } from 'react-router-dom';
import GitHubIcon from '@material-ui/icons/GitHub';
import {
  AppBar,
  Toolbar,
  Button,
  Grid,
  IconButton,
  makeStyles,
  Typography,
} from '@material-ui/core';


const useStyles = makeStyles((theme) => ({
  link: {
    color: 'white',
  },
  title: {
    marginLeft: '10px',
  },
}));

function Footer() {
  const classes = useStyles();

  return (
    <AppBar position='sticky'>
      <Toolbar>
        <Grid container justify='space-between' alignItems='center'>
          <IconButton
            edge='start'
            color='inherit'
            aria-label='github'
            onClick={() => window.open('https://github.com/Strategizer/app')}
          >
            <GitHubIcon color='inherit' />
            <Typography className={classes.title} variant='h6'>
              Github Repository
            </Typography>
          </IconButton>
          <Grid item>
            <Button>
              <Typography variant='button'>
                <Link className={classes.link} to='/contributors'>
                  Our contributors
                </Link>
              </Typography>
            </Button>
          </Grid>
        </Grid>
      </Toolbar>
    </AppBar>
  );
}

export default Footer;
