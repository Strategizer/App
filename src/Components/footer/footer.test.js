import React from 'react';
import Footer from './Footer';
import { cleanup, render } from '@testing-library/react';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';

describe('footer', () => {

  afterEach(cleanup);

  const history = createMemoryHistory();

  const wrapped = (
    <Router history={history}>
      <Footer />
    </Router>
  );

  it('matches the snapshot', () => {
    const { asFragment } = render(wrapped);
    expect(asFragment()).toMatchSnapshot()
  }); 
  

  it('renders required links', () => {
    const { getByText } = render(wrapped);
    const contributorsLink = getByText(/contributors/);
    const repoLink = getByText(/Github Repository/);
    expect(repoLink).toBeTruthy(); 
    expect(contributorsLink).toBeTruthy(); 
  });

});
