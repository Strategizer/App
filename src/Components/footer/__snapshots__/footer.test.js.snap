// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`footer matches the snapshot 1`] = `
<DocumentFragment>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionSticky MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-between"
      >
        <button
          aria-label="github"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 .3a12 12 0 0 0-3.8 23.4c.6.1.8-.3.8-.6v-2c-3.3.7-4-1.6-4-1.6-.6-1.4-1.4-1.8-1.4-1.8-1-.7.1-.7.1-.7 1.2 0 1.9 1.2 1.9 1.2 1 1.8 2.8 1.3 3.5 1 0-.8.4-1.3.7-1.6-2.7-.3-5.5-1.3-5.5-6 0-1.2.5-2.3 1.3-3.1-.2-.4-.6-1.6 0-3.2 0 0 1-.3 3.4 1.2a11.5 11.5 0 0 1 6 0c2.3-1.5 3.3-1.2 3.3-1.2.6 1.6.2 2.8 0 3.2.9.8 1.3 1.9 1.3 3.2 0 4.6-2.8 5.6-5.5 5.9.5.4.9 1 .9 2.2v3.3c0 .3.1.7.8.6A12 12 0 0 0 12 .3"
              />
            </svg>
            <h6
              class="MuiTypography-root makeStyles-title-2 MuiTypography-h6"
            >
              Github Repository
            </h6>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <span
                class="MuiTypography-root MuiTypography-button"
              >
                <a
                  class="makeStyles-link-1"
                  href="/contributors"
                >
                  Our contributors
                </a>
              </span>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </header>
</DocumentFragment>
`;
